### ISSUE TITLE
Why not use regular voting to determine a checkpoint? This could have prevented the 51% attempt

### ISSUE THREAD

**Issue by lostinblue** [ethereumproject/go-ethereum#703]
We could have either some people with special privilege like developers or simply have a time-lock transaction that contains your answer and if its wrong, you lose your money. And x number of votes = the checkpoint gets written to a memory pool or even even hard-coded into Go eventually.

**Comment by lostinblue** [comment]
These are easy fixes we can stop these attacks no problem, and I'm personally going to talk to the miner who did this. 

I'm not nice, if we can't come to an agreement, I will burn down his server farm and make him regret he ever mined ANYTHING or bought a video card to begin with. I have firmware for every major video card and I will burn out every card he ever fucking owns if doesn't act reasonable. And I know for a fucking fact that his little farm is not nearly as secure as he would like it to be. 

I'm not fucking around.

**Comment by mikeyb** [comment]
I have yet to meet a credible hacker who talks about how they are a hacker publicly.  Just saying.  This type of attitude always points to script kiddo

If you have fixes, show with code, not with meaningless threats...  Make a proposal, have it discussed, if it is valid and wanted it will get implemented

**Comment by RorschachRev** [comment]
There are a LOT of big gaping holes in geth. EF version is the main one I look at and work on rewriting. 850k lines of crap.

**Comment by lostinblue** [comment]
You don't know any hackers, what are you talking about. I'm being aggressive make them defy me so I can do the thing I want to do in the first place. 

You have already seen my work but this is the first time you seen me. I've been involved with ETC since the start. 

RorscachRev, thanks I will look into that codebase, you are very right about geth. I dont want to waste time, I just want to optimize it, there are so many ways and get it ready for a project that I have been asked to do. And I need to ensure its not flimsy, so thanks for that great start. 

Like I said, Ill check it out now, put it in a VM, and start working on it immediately.

**Comment by lostinblue** [comment]
Ive grown to like cbor myself

**Comment by lostinblue** [comment]
Oh and not for OHT currently, Gravity rebuilt it completely using segwit, but it wouldn't be hard to include ETC in the new version.

**Comment by lostinblue** [comment]
Lol, well I still have a long list of bugs I can press to shutdown their network again when the time comes, making their clients crash at least or maybe even worse like RCE.

It was funny watching them struggle last time, every fix, I just drop a knew one from my list, and only stopped because I got bored and it proved my point.

**Comment by lostinblue** [comment]
Anyways, time to go line by line and scream, do drugs and make some people happy with a client they can trust for their new project. We will be talking more, nice start.

**Comment by RorschachRev** [comment]
Their protocol layer is spread across so many folders. It is like 30 devs are writing 30 copies of ethereum and as long as it compiles at the end of the day, nobody complains. The layers of cruft have layers of cruft, with obsolete libraries referencing each other in circles.

No, you didn't get fingered as JS dev, I checked your repos. I moved off github a while back because I hate M$. The reason I said that cbor would be good for JS devs is because their format is so bad, node uses cbor fairly well from what I saw, and it is nearly a drop in replacement for them. The idea of having your data structures need updating when your data protocol changes is pretty common. I don't see it being a problem for JS because they don't know what their data format is. They don't have a protocol really. On the other hand, anybody who is coding structs probably sees protobuf as an easy and convenient lack of double work with free protocol schema.

**Comment by RorschachRev** [comment]
Justice is a slippery slope in my experience. I almost considered myself morally just in shutting down 40% of a nations internet access using backbone protocols and an insecurity in their network. I had good motivation, moral high ground... It is a slippery slope. The advice I've given to other people and live by myself is more, "have good aim." If there is collateral damage, you're not morally justified. Employees of a company would count as "collateral damage" in most cases.

I'm working on blockchain stuff atm, I want to get together a security team specifically for blockchain. Shit like IOTA just isn't permitted imho. Credits.com is 10x worse, they don't even nonce. Their whitepaper claims that they don't need to prevent double spend, and then they call themselves a secure escrow service. Even worse their investors and advisors say it doesn't matter, because they make too much money off the bad code.

**Comment by lostinblue** [comment]
Just got like 10x cooler dude.

**Comment by RorschachRev** [comment]
No, that wasn't me. I was shutting down a guy in brazil due to forcing a kind of dumb 15 year old on webcam. I hacked the same server he did to get the user+pass list from his bad phishing site, then had the people who asked for my help message every user and warn them about this guy while he was inexplicably offline. Reported the guy, reported the server, reported the phishing. Did what I could without collateral damage. I wouldn't run IRC networks, too many ways to break them. Neither 2600 nor Anonymous could keep an IRC network up, they just took turns hacking them.

Look me up somewhere, same nick.
